<a href="https://colab.research.google.com/github/nenecisse0524/bank-ml/blob/main/PROJECT_4_COVID_19_MACHINE_LEARNING.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

                                         PROBLEM STATEMENT
    
   The Data is related to the ICU visit Prediction during COVID 19. The HOSPITAL is SÍRIO-LIBANÊS in Brazil one of the countries most affected by the COVID 19 pandemic with more than 16 million confirmed cases and 454 429 confirmed deaths as of May 26, 2021.we willuse Machine Learning ML to help reduce the strain on hospital's ICU beds, where the objective is to develop a ML model to predict if a patient of a confirmed COVID-19 case will require admission to the ICU.

In this assignment, you are challenged to perform a full lifecycle ML model development according to the objective of the dataset, which includes the following elements:

Conduct EDA (Exploratory Data Analysis) to gain an understanding of the Data
Preprocess the Data to prepare it for the ML Model
Develop a few candidate ML models and pick the most promising one. Evaluate it based on a variety of metrics as well as analysing it’s performance (does it overtrain, can we improve it by changing our data preprocessing etc.)
Deliverables:
Upload your code to a github repository as a Jupyter Notebook. Use the commit system to update your progress whenever you are working on the progress (rather than just upload the file at the end). Ensure you have a readme that explains what you are doing.
Executive Presentation: Prepare a presentation for the board of executives where you explain why your model can help save lives, why the hospital should start using it and present it’s performance as well as trends in the data you discovered during EDA.
Research Report: Write a paper with up to 2000 words (this can be less as long as the important information is included) where you present your methods and finding to an expert audience. Your paper should roughly follow the structure:
Introduction (why are we analysing the data/What is the problem)
Data Description (describe the data you are working with in detail)
Methods (what methods are you using for your research)
Machine Learning model development (what decisions did you make, why?)
Performance analysis
Conclusion

Let's start by importing all the packages we will need.
pandas is for data analysis
Matplotlib is for visualisation
Seaborn is for some advance data analysis tolls
numpy is for numerical
sklearn is for linear regression

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import statsmodels.api as sm
import sklearn as sl
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LogisticRegression
from sklearn.metrics import accuracy_score,confusion_matrix,classification_report
from sklearn.preprocessing import LabelEncoder
from sklearn.preprocessing import StandardScaler
import warnings
warnings.filterwarnings('ignore')

1) Let's do our EDA (Exploratory Data Analysis) and it will help us better understand the Data.

In [3]:
covid = pd.read_excel('/content/Kaggle_Sirio_Libanes_ICU_Prediction (1).xlsx')
covid.head()

PATIENT_VISIT_IDENTIFIER  AGE_ABOVE65 AGE_PERCENTIL  GENDER  \
0                         0            1          60th       0   
1                         0            1          60th       0   
2                         0            1          60th       0   
3                         0            1          60th       0   
4                         0            1          60th       0   

   DISEASE GROUPING 1  DISEASE GROUPING 2  DISEASE GROUPING 3  \
0                 0.0                 0.0                 0.0   
1                 0.0                 0.0                 0.0   
2                 0.0                 0.0                 0.0   
3                 0.0                 0.0                 0.0   
4                 0.0                 0.0                 0.0   

   DISEASE GROUPING 4  DISEASE GROUPING 5  DISEASE GROUPING 6  ...  \
0                 0.0                 1.0                 1.0  ...   
1                 0.0                 1.0                 1.0  ...   
2                 0.0                 1.0                 1.0  ...   
3                 0.0                 1.0                 1.0  ...   
4                 0.0                 1.0                 1.0  ...   

   TEMPERATURE_DIFF  OXYGEN_SATURATION_DIFF  BLOODPRESSURE_DIASTOLIC_DIFF_REL  \
0         -1.000000               -1.000000                         -1.000000   
1         -1.000000               -1.000000                         -1.000000   
2               NaN                     NaN                               NaN   
3         -1.000000               -1.000000                               NaN   
4         -0.238095               -0.818182                         -0.389967   

   BLOODPRESSURE_SISTOLIC_DIFF_REL  HEART_RATE_DIFF_REL  \
0                        -1.000000            -1.000000   
1                        -1.000000            -1.000000   
2                              NaN                  NaN   
3                              NaN                  NaN   
4                         0.407558            -0.230462   

   RESPIRATORY_RATE_DIFF_REL  TEMPERATURE_DIFF_REL  \
0                  -1.000000             -1.000000   
1                  -1.000000             -1.000000   
2                        NaN                   NaN   
3                        NaN             -1.000000   
4                   0.096774             -0.242282   

   OXYGEN_SATURATION_DIFF_REL    WINDOW  ICU  
0                   -1.000000       0-2    0  
1                   -1.000000       2-4    0  
2                         NaN       4-6    0  
3                   -1.000000      6-12    0  
4                   -0.814433  ABOVE_12    1  

[5 rows x 231 columns]

In [4]:
print("size/shape of dataset" ,covid.shape)
print("checking for null values" ,covid.isnull().sum())
print("checking for duplicate values" ,covid.duplicated().sum())
print("checking for data types" ,covid.dtypes)

size/shape of dataset (1925, 231)
checking for null values PATIENT_VISIT_IDENTIFIER        0
AGE_ABOVE65                     0
AGE_PERCENTIL                   0
GENDER                          0
DISEASE GROUPING 1              5
                             ... 
RESPIRATORY_RATE_DIFF_REL     748
TEMPERATURE_DIFF_REL          694
OXYGEN_SATURATION_DIFF_REL    686
WINDOW                          0
ICU                             0
Length: 231, dtype: int64
checking for duplicate values 0
checking for data types PATIENT_VISIT_IDENTIFIER        int64
AGE_ABOVE65                     int64
AGE_PERCENTIL                  object
GENDER                          int64
DISEASE GROUPING 1            float64
                               ...   
RESPIRATORY_RATE_DIFF_REL     float64
TEMPERATURE_DIFF_REL          float64
OXYGEN_SATURATION_DIFF_REL    float64
WINDOW                         object
ICU                             int64
Length: 231, dtype: object


The size and shape show that we have 1925 rows and 231 columns.
we have null values in disease grouping 1 showing 5.
As for the data type we have int and float.
The lenght is 231